org/postgresql/PGProperty.java

```diff
@@ -204,6 +204,16 @@
      * is specified in seconds and a value of zero means that it is disabled.
      */
     SOCKET_TIMEOUT("socketTimeout", "0", "The timeout value used for socket read operations."),
+    
+    /**
+     * Socket factory used to create socket. A null value, which is the default, means system default.
+     */
+    SOCKET_FACTORY("socketFactory", null, "Specify a socket factory for socket creation"),
+    
+    /**
+     * The String argument to give to the constructor of the Socket Factory
+     */
+    SOCKET_FACTORY_ARG("socketFactoryArg", null, "Argument forwarded to constructor of SocketFactory class."),
```

Comment:
@Albatarm , could you please clarify why that was needed?

Apparently, `socketFactory` receives the whole `Properties` object.
Why pollute public API with `socketFactoryArg`?

Reply:
I just copied the form of `SSL_FACTORY` and `SSL_FACTORY_ARG`.